src/Illuminate/Foundation/Testing/DatabaseTransactionsManager.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Illuminate\Foundation\Testing;
+
+class DatabaseTransactionsManager extends \Illuminate\Database\DatabaseTransactionsManager
+{
+    /**
+     * Register a transaction callback.
+     *
+     * @param  callable  $callback
+     * @return void
+     */
+    public function addCallback($callback)
+    {
+        // If there are no transactions, we'll run the callbacks right away. Also, we'll run it
+        // right away when we're in test mode and we only have the wrapping transaction. For
+        // every other case, we'll queue up the callback to run after the commit happens.
+        if ($this->callbackApplicableTransactions()->count() === 0) {
+            return $callback();
+        }
+
+        $this->transactions->last()->addCallback($callback);
+    }
+
+    /**
+     * Get the transactions that are applicable to callbacks.
+     *
+     * @return \Illuminate\Support\Collection<int, \Illuminate\Database\DatabaseTransactionRecord>
+     */
+    public function callbackApplicableTransactions()
+    {
+        return $this->transactions->reject(
+            fn ($transaction) => $transaction === $this->transactions->first()
+        )->values();
```

Comment:
Could be re-written like this, right?!

```suggestion
        return $this->transactions->skip(1)->values();
```